osdep/timer.c

```diff
@@ -87,22 +97,16 @@ static int get_realtime(struct timespec *out_ts)
     return 0;
 #endif
 }
-#endif
 
 struct timespec mp_time_ns_to_realtime(int64_t time_ns)
 {
     struct timespec ts = {0};
-
-#if !HAVE_WIN32_INTERNAL_PTHREADS
     if (get_realtime(&ts) != 0)
         return ts;
-    int64_t time_rel = time_ns - mp_time_ns();
-#else
-    int64_t time_rel = time_ns;
-#endif
 
     // clamp to 1000 days in the future
-    time_rel = MPMIN(time_rel, 1000 * 24 * 60 * 60 * INT64_C(1000000000));
+    int64_t time_rel = MPMIN(mp_time_ns() - time_ns,
```

Comment:
You flipped the order, it will not work, not only on Windows.

Should be `time_ns - mp_time_ns()`

Reply:
`MPMIN` also can end up evaluating the argument twice. Better hoist the function call out.

Reply:
If we want to get concerned about double evaluations, it's probably better to change the implementation of MPMIN and MPMAX to use statement expressions if possible. That be a free win everywhere the macro is used (which is a ton a place).

Reply:
I didn't write this code, it's merely a revert of the win32-specific commit.
So that commit also contained a fix applicable to all platforms? Not ideal, again.

Reply:
> although it seems that would introduce a ton of problems doing some research. thanks GNU

I don't think it's GNU's fault for adding useful shit. In fact, ISO C usually requires there to be (AFAIK at least 2) practical implementations to be present before including something in the standard. So if no one was writing extensions, then nothing would ever improve.

Reply:
fixed.

Reply:
>  So that commit also contained a fix applicable to all platforms? Not ideal, again.

Yes, so? If I made a PR I test it as a whole, not individual commits. Especially when amending fixes. If I had luxury of making smaller PR, I would, but the turnaround even for small changes is the same, so I would stagger myself to death.

I missed that, and probably squashed to the later commit, instead of the former one. Same as you missed that during the review and revert. No big deal, happens...

Could we stop with those "not ideal", "commit msg" bad nitpicks? I acknowledged that, move along.